Sources/swift-format/Frontend/Frontend.swift

```diff
@@ -124,27 +124,42 @@ class Frontend {
   }
 
   /// Processes source content from a list of files and/or directories provided as paths.
-  private func processPaths(_ paths: [String]) {
+  private func processPaths(_ paths: [String], parallel: Bool) {
     precondition(
       !paths.isEmpty,
       "processPaths(_:) should only be called when paths is non-empty.")
 
-    for path in FileIterator(paths: paths) {
+    let concurrentQueue = DispatchQueue(
+        label: "swift-format-queue", qos: .userInitiated, attributes: .concurrent)
+    let lock = NSLock()
+    let openAndProcess: (String) -> Void = { path in
       guard let sourceFile = FileHandle(forReadingAtPath: path) else {
-        diagnosticEngine.diagnose(Diagnostic.Message(.error, "Unable to open \(path)"))
-        continue
+        lock.lock()
+        defer { lock.unlock() }
```

Comment:
Note this is on both paths but I assume that's ok since this is an unexpected case anyways

Reply:
There's this diagnostic, but then there's all the others. Have you tried parallel mode with the `lint` command yet, especially on a large set of files that have a large number of warnings?

SwiftSyntax's `DiagnosticEngine` and `PrintingDiagnosticConsumer` don't have any synchronization that I can see, and in particular, `PrintingDiagnosticConsumer.handle` makes [multiple calls](https://github.com/apple/swift-syntax/blob/master/Sources/SwiftSyntax/PrintingDiagnosticConsumer.swift#L39-L54) to `FileHandle.write`, so unless I've missed something, I'd be surprised if this doesn't interleave diagnostics.

If we're going to support a parallel mode, I think we either need to provide our own diagnostic consumer that synchronizes its `handle` method, or make SwiftSyntax's `PrintingDiagnosticConsumer` do it by default. @akyrtzi

Reply:
Added the lock to the args of this and only passed it in the concurrent case